src/docs/testing/best-practices.md

```diff
@@ -0,0 +1,173 @@
+
+* [ShaderMask]({{site.api}}/flutter/widgets/ShaderMask-class.html)
+* [ColorFilter]({{site.api}}/flutter/dart-ui/ColorFilter-class.html)
+* [Chip]({{site.api}}/flutter/material/Chip-class.html)&mdash;may
+  cause call to `saveLayer()` if `disabledColorAlpha != 0xff`
+* [Text]({{site.api}}/flutter/widgets/Text-class.html)&mdash;may
+  cause call to `saveLayer()` if there's an `overflowShader`
+
+Ways to avoid calls to `saveLayer()`:
+
+* To implement fading in an image, consider using the FadeInImage widget,
+  which applies a gradual opacity using the GPU’s fragment shader.
+  For more information, see
+  [Opacity]({{site.api}}/flutter/widgets/Opacity-class.html#transparent-image).
+  ([PENDING] not released yet)
+* To create a rectangle with rounded corners, instead of applying a
+  clipping rectangle, consider using the `borderRadius` property offered
+  by many of the widget classes.
+
+### Render grids and lists lazily
+
+Use the lazy methods, with callbacks, when building large grids or lists.
+That way only the visible portion of the screen is built at startup time.
+
+Also see:
+
+* [Working with long lists](/cookbook/lists/long-lists) in the
+  [Cookbook](/cookbook)
+* [Creating a ListView that loads one page at a
+  time](https://medium.com/saugo360/flutter-creating-a-listview-that-loads-one-page-at-a-time-c5c91b6fabd3)
+  by AbdulRahman AlHamali
+* [Listview.builder]({{site.api}}/flutter/widgets/ListView/ListView.builder.html) API
+
+###  Build and display frames in 16ms
+
+Since there are two separate threads for building and rendering, you
+have 16ms for building, and 16ms for rendering on a 60Hz display.
+If latency is a concern, build and display a frame in 16ms _or less_.
+Note that means built in 8ms or less,
+and rendered in 8ms or less, for a total of 16ms or less.
+If missing frames (jankyness) is a concern, then 16ms for each of
+the build and render stages is OK.
+
+If your frames are rendering in well under 16ms total in a [profile
+build](/docs/testing/build-modes#profile),
+you likely don’t have to worry about performance even if some
+performance pitfalls apply, but you should still aim to build and
+render a frame as fast as possible. Why?
+
+* Lowering the frame render time below 16ms may not make a visual
+  difference, but it will **improve battery life** and thermal issues.
+* It may run fine on your device, but consider performance for the
+  lowest device you are targeting.
+* When 120fps devices become widely available, you’ll want to render frames
+  in under 8ms (total) in order to provide the smoothest experience.
+
+If you are wondering why 60fps leads to a smooth visual experience,
+see the video [Why 60fps?](https://www.youtube.com/watch?v=CaMTIgxCSqU)
+
+## Pitfalls
+
+If you need to tune your app’s performance, or perhaps the UI isn't as
+smooth as you expect, the Flutter plugin for your IDE can help.
+In the Flutter inspector, bring up the **Performance** tab and enable the
```

Comment:
The performance tab is no longer in the inspector. It has its own tool window.

Perhaps, "In the Flutter Performance window, enable the **Show widget rebuild information** check box."

<img width="847" alt="screen shot 2019-01-14 at 8 23 36 am" src="https://user-images.githubusercontent.com/43759233/51125527-b5f95500-17d5-11e9-852b-56bd8ee17492.png">